pkg/registry/apis/datasource/plugincontext.go

```diff
@@ -2,10 +2,122 @@ package datasource
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/grafana/grafana/pkg/apis/datasource/v0alpha1"
+	"github.com/grafana/grafana/pkg/infra/appcontext"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/grafana-apiserver/endpoints/request"
+	"github.com/grafana/grafana/pkg/services/grafana-apiserver/utils"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/plugincontext"
+)
+
+// This provides access to settings saved in the database.
+// Authorization checks will happen within each function, and the user in ctx will
+// limit which namespace/tenant/org we are talking to
+type PluginConfigProvider interface {
```

Comment:
I'm a bit late to the party but what are we calling this `ConfigProvider` when it's returning data sources?

It would be a bit more dry if this was:

```
type PluginDatasourceProvider interface {
  Get(...)
  List(...)
  GetInstanceSettings(...)
}
```

or am I missing something?

Reply:
We can change that as we see fit